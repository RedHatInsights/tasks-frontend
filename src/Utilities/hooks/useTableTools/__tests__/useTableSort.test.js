import { act, renderHook } from '@testing-library/react-hooks';
import useTableSort, { useTableSortWithItems } from '../useTableSort';
import { taskColumnsFixtures as columns } from './__fixtures__/columns.fixtures';

describe('useTableSort', () => {
  it('returns a table sort configuration', () => {
    const { result } = renderHook(() => useTableSort(columns));
    expect(result).toMatchSnapshot();
  });

  it('returns a table sort configuration with an inital state', () => {
    const sortBy = {
      index: 3,
      direction: 'asc',
    };
    const { result } = renderHook(() =>
      useTableSort(columns, {
        sortBy,
      })
    );
    expect(result.current.tableProps.sortBy).toEqual(sortBy);
  });

  it('updates sortBy with onSort', () => {
    const sortBy = {
      index: 3,
      direction: 'asc',
    };
    const newSortBy = {
      index: 0,
      direction: 'desc',
    };

    const { result } = renderHook(() =>
      useTableSort(columns, {
        sortBy,
      })
    );

    act(() => {
      result.current.tableProps.onSort(null, 0, 'desc');
    });

    expect(result.current.tableProps.sortBy).toEqual(newSortBy);
  });
});

describe('useTableSortWithItems', () => {
  it('returns no sortBy when there are no items', () => {
    const sortBy = {
      index: 3,
      direction: 'asc',
    };
    const { result } = renderHook(() =>
      useTableSortWithItems([], columns, {
        sortBy,
      })
    );
    expect(result.current.tableProps.sortBy).toEqual(undefined);
  });
});
