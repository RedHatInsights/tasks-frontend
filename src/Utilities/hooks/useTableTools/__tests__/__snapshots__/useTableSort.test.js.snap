// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTableSort returns a table sort configuration 1`] = `
Object {
  "all": Array [
    Object {
      "sorter": [Function],
      "tableProps": Object {
        "cells": Array [
          Object {
            "props": Object {
              "width": 35,
            },
            "sortByProp": "task_title",
            "title": "Task",
            "transforms": Array [
              [Function],
            ],
          },
          Object {
            "props": Object {
              "width": 20,
            },
            "renderExport": [Function],
            "sortByProp": "systems_count",
            "title": "Systems",
            "transforms": Array [
              [Function],
            ],
          },
          Object {
            "props": Object {
              "width": 20,
            },
            "sortByProp": "end_time",
            "title": "Run date/time",
            "transforms": Array [
              [Function],
            ],
          },
        ],
        "onSort": [Function],
        "sortBy": Object {
          "direction": "asc",
          "index": 3,
        },
      },
    },
  ],
  "current": Object {
    "sorter": [Function],
    "tableProps": Object {
      "cells": Array [
        Object {
          "props": Object {
            "width": 35,
          },
          "sortByProp": "task_title",
          "title": "Task",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "props": Object {
            "width": 20,
          },
          "renderExport": [Function],
          "sortByProp": "systems_count",
          "title": "Systems",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "props": Object {
            "width": 20,
          },
          "sortByProp": "end_time",
          "title": "Run date/time",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "onSort": [Function],
      "sortBy": Object {
        "direction": "asc",
        "index": 3,
      },
    },
  },
  "error": undefined,
}
`;
